### PR TITLE
Feat:  eth_simulateV1 Request / Response types

### PR DESCRIPTION
**Pull request by DoTheBestToGetTheBest**
I'am going to first convert this to draft, i need some help on some type, like error one and back & stateDb should i use Revm and foundry backend here or i do have to manually implemnt those ? Thank you 

This pr should relate https://github.com/alloy-rs/alloy/issues/901

### CODE REVIEW

File: crates/rpc-types-trace/src/simulate.rs [hunk 1]
```diff
@@ -0,0 +1,75 @@
+//! 'eth_simulateV1' Request / Response types
+
+use alloy_primitives::{Bytes, Log, B256, U64};
+use alloy_rpc_types_eth::{state::StateOverride, BlockOverrides, Header, TransactionRequest};
+use serde::{Deserialize, Serialize};
+
+/// The maximum number of blocks that can be simulated in a single request,
+pub const MAX_SIMULATE_BLOCKS: U64 = U64::from_limbs([256_u64]);
```
Comment (mattsse) [inline crates/rpc-types-trace/src/simulate.rs:8]: this should be a regular u64

File: crates/rpc-types-trace/src/simulate.rs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+#[serde(rename_all = "camelCase")]
+pub struct SimCallResult {
+    /// The raw bytes returned by the transaction.
+    pub return_value: Bytes,
+    /// Logs generated during the execution of the transaction.
+    pub logs: Vec<Log>,
+    /// The amount of gas used by the transaction.
+    pub gas_used: U64,
+    /// The final status of the transaction, typically indicating success or failure.
+    pub status: U64,
```
Comment (mattsse) [inline crates/rpc-types-trace/src/simulate.rs:36]: let's use u64 and `serde(with alloy_serde::quantity`

File: crates/rpc-types-trace/src/simulate.rs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    /// Flag to determine whether to trace ERC20/ERC721 token transfers within transactions.
+    pub trace_transfers: bool,
+    /// Flag to enable or disable validation of the transaction sequence in the blocks.
+    pub validation: bool,
+    /// Flag to decide if full transactions should be returned instead of just their outcomes.
+    pub return_full_transactions: bool,
+}
+
+/// Represents a simulator backend to handle state and transaction processing.
+#[derive(Clone, Debug)]
```
Comment (mattsse) [inline crates/rpc-types-trace/src/simulate.rs:58]: needs camelcase

**Review by mattsse — CHANGES_REQUESTED** [review]
cool great start

**Comment by DoTheBestToGetTheBest** [comment]
> cool great start
> 
> >

ty!




> those will be implemented by the nodes itself

ty but i'm not sure to understand wdym by this ?

**Review by mattsse — CHANGES_REQUESTED** [review]
in this pr we're only interested in the types

could you perhaps add some sanity serde json tests with sample requests from

https://github.com/ethereum/execution-apis/pull/484/files#diff-358634b5754d6188b1888d3bb89dbe4c8ba62ea941f2afa4d58e2a8a3bdf0f41

and we should move this file to the rpc-types-eth crate

File: crates/rpc-types-trace/src/simulate.rs [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+/// Represents a simulator backend to handle state and transaction processing.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Simulator {
+    //What we should use here for backend ?
+    //b              Backend
+    /// List of hashes representing the blocks to be simulated.
+    pub hashes: Vec<B256>,
+    // should we use StadeDB of revm here ?
```
Comment (Blaeaea) [inline crates/rpc-types-trace/src/simulate.rs:66]: @mattsse please is there a way to know if it's needed to use revm StateDB here?

File: crates/rpc-types-trace/src/simulate.rs [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    pub validation: bool,
+    /// Flag to decide if full transactions should be returned instead of just their outcomes.
+    pub return_full_transactions: bool,
+}
+
+/// Represents a simulator backend to handle state and transaction processing.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Simulator {
+    //What we should use here for backend ?
```
Comment (Blaeaea) [inline crates/rpc-types-trace/src/simulate.rs:62]: @mattsse would be great if we also know if it's possible to use foundry backend here

**Comment by mattsse** [comment]
@DoTheBestToGetTheBest do you have the bandwidth to finalize this pr?

we'd need a few tests that serialize data from the examples
https://github.com/ethereum/execution-apis/pull/484/files

**Comment by DoTheBestToGetTheBest** [comment]
> @DoTheBestToGetTheBest do you have the bandwidth to finalize this pr?
> 
> we'd need a few tests that serialize data from the examples https://github.com/ethereum/execution-apis/pull/484/files

sorry, it was already done and i though i commited change

File: crates/rpc-types-trace/src/simulate.rs (hunk 4 above)
Comment (DoTheBestToGetTheBest) [reply crates/rpc-types-trace/src/simulate.rs]: this going to be implemented in reth

File: crates/rpc-types-trace/src/simulate.rs (hunk 5 above)
Comment (DoTheBestToGetTheBest) [reply crates/rpc-types-trace/src/simulate.rs]: same

**Review by mattsse — APPROVED** [review]
cool,

this is likely not entirely complete what will unblock reth work to figure out what's missing

**Pull request merged**